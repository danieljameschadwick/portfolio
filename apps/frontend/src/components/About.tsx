import { Attributes } from "../components/Attributes/Attributes";

export const About: React.FC = () => {
  return (
    <>
      <div className={"about-container initial-container"}>
        <div className={"about-content container px-16 pb-8 sm:px-12 sm:pb-14"}>
          <h2 className={"heading"}>About Me</h2>

          <p>
            Developer at i6 working with React, JavaScript, LESS,
            Twig/HTML/Mustache, REST APIs and MongoDB. Also experienced using
            Doctrine ORM, MySQL 8.0, PHPUnit, Behat, RabbitMQ, AWS, Google
            Cloud, Docker, and Elasticsearch.
          </p>

          <p>
            I studied at the University of Hull for my BSc Computer Science.
            Whilst at University I mainly used C# but have developed in PHP,
            Java, C++, C and JavaScript and studied an array of key concepts
            such as MVC frameworks, data structures, and algorithms to name a
            few.
          </p>

          <p>
            Prior to studying I worked for 2 years at an online eCommerce
            retailer as a Full Stack Developer focusing on PHP
            eCommerce/management systems, scripts and software environments.
          </p>

          <p>Some of my work is featured further down.</p>
        </div>
      </div>

      <Attributes />
    </>
  );
};
