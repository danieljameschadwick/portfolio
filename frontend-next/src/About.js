import { Attributes } from "./Attributes/Attributes";

export const About = () => {
    return (
        <>
            <div className={"about-container initial-container"}>
                <div className={"about-content px-16 pb-8 sm:px-12 sm:pb-14"}>
                    <h2 className="heading">About Me</h2>

                    <p>
                        Developer at Wren Kitchens working with mainly PHP7/Symfony Framework, LESS/CSS, JavaScript,
                        Twig/HTML, MySQL on a variety of internal and external services. Also experienced using Doctrine ORM,
                        PHPUnit, Behat, RabbitMQ, AWS, Docker, and Elasticsearch.
                    </p>

                    <p>
                        I studied at the University of Hull for my BSc Computer Science. Whilst at University I mainly
                        used C# but have developed in PHP, Java, C++, C and JavaScript and studied an array of key concepts such
                        as MVC frameworks, data structures, and algorithms to name a few.
                    </p>

                    <p>
                        Prior to studying I worked for 2 years at an online eCommerce retailer as a Full Stack Developer
                        focusing on PHP eCommerce/management systems, scripts and software environments.
                    </p>

                    <p>Some of my work is featured further down.</p>
                </div>
            </div>

            <Attributes />
        </>
    );
};
