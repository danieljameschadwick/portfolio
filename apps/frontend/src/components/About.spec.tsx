import { render } from "@testing-library/react";
import { About } from "./About";

describe("About component", () => {
  it("renders base", () => {
    const { baseElement } = render(<About />);

    expect(baseElement).toMatchSnapshot();
  });
});
