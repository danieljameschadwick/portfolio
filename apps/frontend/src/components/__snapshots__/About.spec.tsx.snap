// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders base 1`] = `
<body>
  <div>
    <div
      class="about-container initial-container"
    >
      <div
        class="about-content container px-16 pb-8 sm:px-12 sm:pb-14"
      >
        <h2
          class="heading"
        >
          About Me
        </h2>
        <p>
          Software Developer at i6 working with React, JavaScript/TypeScript, Jest, React Testing Library, Redux, LESS, i18n. Also experienced using PHP, Symfony, Doctrine ORM, MySQL 8.0, MongoDB, PHPUnit, Behat, CircleCI, RabbitMQ, AWS, Google Cloud, and Docker.
        </p>
        <p>
          I studied at the University of Hull for my BSc Computer Science. Whilst at University I mainly used C# but have developed in PHP, Java, C++, C and JavaScript and studied an array of key concepts such as MVC frameworks, data structures, and algorithms to name a few.
        </p>
        <p>
          Prior to studying I worked for 2 years at an online eCommerce retailer as a Full Stack Developer focusing on PHP eCommerce/management systems, scripts and software environments.
        </p>
        <p>
          Some of my work is featured further down.
        </p>
      </div>
    </div>
    <div
      class="attributes-container container mt-12 mb-10 mx-auto w-9/12"
    >
      <div
        class="attributes"
      >
        <div
          class="attribute"
        >
          <span
            class="icon"
          >
            <div>
              <div
                data-src="img/svg/attribute/frontend.svg"
                src=""
              />
            </div>
          </span>
          <h2
            class="heading"
          >
            Frontend
          </h2>
          <p
            class="text"
          >
            Specialising in response, modern, designs which translate to any business ranging from small businesses to large enterprises.
          </p>
          <h3
            class="heading"
          >
            Skills
          </h3>
          <ul
            class="list"
          >
            <li>
              React
            </li>
            <li>
              HTML/CSS
            </li>
            <li>
              LESS/SASS
            </li>
            <li>
              JavaScript
            </li>
            <li>
              TailwindCSS
            </li>
            <li>
              Twig
            </li>
          </ul>
          <h3
            class="heading"
          >
            Studying
          </h3>
          <ul
            class="list"
          >
            <li>
              NextJS
            </li>
          </ul>
        </div>
        <div
          class="attribute"
        >
          <span
            class="icon"
          >
            <div>
              <div
                data-src="img/svg/attribute/backend.svg"
                src=""
              />
            </div>
          </span>
          <h2
            class="heading"
          >
            Backend
          </h2>
          <p
            class="text"
          >
            Responsible for developing applications which can scale to the masses - for both internal users or public. Experienced in developing standalone applications with cached data, microservices and monolithic applications.
          </p>
          <h3
            class="heading"
          >
            Skills
          </h3>
          <ul
            class="list"
          >
            <li>
              PHP7
            </li>
            <li>
              Symfony Framework 3/4
            </li>
            <li>
              Doctrine ORM
            </li>
            <li>
              RESTful APIs
            </li>
            <li>
              OOP, SOLID, MVC
            </li>
            <li>
              PSR
            </li>
          </ul>
          <h3
            class="heading"
          >
            Studying
          </h3>
          <ul
            class="list"
          >
            <li>
              NestJS
            </li>
            <li>
              MikroORM
            </li>
          </ul>
        </div>
        <div
          class="attribute"
        >
          <span
            class="icon"
          >
            <div>
              <div
                data-src="img/svg/attribute/devops.svg"
                src=""
              />
            </div>
          </span>
          <h2
            class="heading"
          >
            DevOps
          </h2>
          <p
            class="text"
          >
            Developing web applications using Amazon S3, CloudWatch and CloudFront.
          </p>
          <h3
            class="heading"
          >
            Skills
          </h3>
          <ul
            class="list"
          >
            <li>
              AMQP (RabbitMQ)
            </li>
            <li>
              Docker
            </li>
            <li>
              Apache/nginx
            </li>
            <li>
              Linux (CentOS, Ubuntu)
            </li>
            <li>
              Git
            </li>
          </ul>
          <h3
            class="heading"
          >
            Studying
          </h3>
          <ul
            class="list"
          >
            <li>
              Advanced AMQP
            </li>
            <li>
              AWS Lambda
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</body>
`;
